app/assets/javascripts/box_selector.js.jsx

```diff
@@ -14,30 +17,51 @@ var BoxSelector = React.createClass({
     const missingQuantity = selectedBoxes.length - haveQc
     return (
       <div>
-        {haveQc > 0 && this.includeQcInfoCheckbox()}
-        {missingQuantity > 0 && this.qcInfoMessage(missingQuantity, selectedBoxes)}
+        {haveQc > 0 && this.state.displayQcInfo && this.includeQcInfoCheckbox()}
+        {missingQuantity > 0 && this.state.displayQcInfo && this.qcInfoMessage(missingQuantity, selectedBoxes)}
       </div>
     )
   },
 
   renderBoxes: function() {
     const selector = this;
-
+    
     function renderBox(box, i) {
       function handleClick(e) {
         e.preventDefault();
         selector.removeBox(box);
       }
 
       let id = Math.floor(Math.random() * 1000000000);
```

Comment:
The id with a random number? It is used by this component on transfer packages to allow multiple transfers. Was implemented no long ago in [this](https://github.com/instedd/cdx/commit/444317679b13531001f10c37a9dd5f28d2caa802) commit. I don't dare to change it.

Reply:
@straight-shoota do you recall something about that commit Lean is mentioning?

Reply:
That commit actually just changed the unique id generation from `Date.now()` to `Math.random()`.
IIRC `Date.now()` has millisecond precision and two consecutive interactions can produce the same timestamp. Thus `Date.now()` is not a great generator for unique IDs. This broke some of the tests (because test driver interactions are quick). So I changed it to use `Math.random()`. This isn't very sophisticated but avoids collisions with a very high probability.
This is the commit in isolated context: https://github.com/instedd/cdx/commit/444317679b13531001f10c37a9dd5f28d2caa802

Reply:
Right, that makes perfect sense. I still don't quite understand why we need unique IDs here. That was the intention of my original observation. Nevermind, I was curious.